resources/js/processes/modeler/components/inspector/ConfigEditor.vue

```diff
@@ -21,32 +21,78 @@
       />
     </div>
     <small class="form-text text-muted">{{ $t(helper) }}</small>
-    <b-modal
-      v-cloak
-      v-model="showPopup"
-      size="lg"
-      centered
-      :title="$t('Script Config Editor')"
-      header-close-content="&times;"
+    <b-sidebar
+      id="tree-view-sidebar"
+      ref="sidebar"
+      class="sidebar-tree-view"
+      sidebar-class="border"
+      no-header
+      right
     >
-      <div class="editor-container">
-        <monaco-editor
-          v-model="code"
-          :options="monacoLargeOptions"
-          language="json"
-          class="editor"
-          @focusout.native="handleBlur"
-        />
-      </div>
-      <div slot="modal-footer">
-        <b-button
-          class="btn btn-secondary"
-          @click="closePopup"
-        >
-          {{ $t('Close') }}
-        </b-button>
-      </div>
-    </b-modal>
+      <b-card no-body>
+        <b-card-text>
+          <div class="sidebar-header">
+            <b-row>
+              <b-col md="auto">
+                <div class="sidebar-title">
+                  <h6>{{ $t('Data Browser') }}</h6>
+                </div>
+              </b-col>
+              <b-col class="text-right">
+                <div class="mb-2 custom-control custom-switch">
+                  <input id="tree_view" type="checkbox" v-model="treeView"  class="custom-control-input">
+                  <label for="tree_view" class="custom-control-label">{{ $t('Tree View') }}</label>
+                </div>
+              </b-col>
+              <b-col md="auto" class="text-right">
+                <button
+                  class="close"
+                  @click="closeSidebar"
+                >
+                  ×
+                </button>
+              </b-col>
+            </b-row>
+          </div>
+        </b-card-text>
+      </b-card>
+      <b-row>
+        <b-col>
+          <b-card-group>
+            <b-card no-body>
+              <template #header>
+                <div class="sidebar-subtitle">
+                  <h6>{{ $t('Preview screen') }}</h6>
+                </div>
+              </template>
+              <b-card-body>
+                <div class="editor-container">
+                  <monaco-editor
+                    v-model="code"
+                    :options="monacoLargeOptions"
+                    language="json"
+                    class="editor"
+                    @focusout.native="handleBlur"
+                  />
+                </div>
+              </b-card-body>
+            </b-card>
+            <b-card v-if="treeView" no-body>
+              <template #header>
+                <div class="sidebar-subtitle">
+                  <h6>{{ $t('Tree View') }}</h6>
+                </div>
+              </template>
+              <b-card-text>
+                <div class="editor-container">
+                  <tree-view v-model="code" style="border:1px; solid gray; min-height:700px;"></tree-view>
```

Comment:
@gustavobascope Do you think it's possible to have this reactive so it updates the tree when it changes (and is valid JSON)? In this side-by-side view I think it would great if it could update without having to close and re-open the tree view.